### PR TITLE
Js2

### PR DESCRIPTION
**Pull request by Mariannebp**

### CODE REVIEW

**Comment by Fermain** [comment]
Issue - User can attempt to create a new post on the homepage before they have logged in, creating a server error `Malformed Token`. Please resolve this before the review can continue.

**Comment by Mariannebp** [comment]
Should be fixed now.

File: js/handlers/createPost.mjs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (form) {
+    form.addEventListener("submit", (event) => {
+      event.preventDefault();
+
+      const title = newTitle.value;
+      const text = newText.value;
+      const media = newMedia.value;
+
+      createPost({
```
Comment (Fermain) [inline js/handlers/createPost.mjs]:
```suggestion
      await createPost({
```

File: js/handlers/createPost.mjs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * A listener for when creating new posts
+ */
+export function setCreateNewPostListener() {
+  const form = document.querySelector("#addNewPost");
+  const newTitle = document.querySelector("#title");
+  const newText = document.querySelector("#text");
+  const newMedia = document.querySelector("#media");
+
+  if (form) {
+    form.addEventListener("submit", (event) => {
```
Comment (Fermain) [inline js/handlers/createPost.mjs]:
```suggestion
    form.addEventListener("submit", async (event) => {
```

File: js/handlers/createPost.mjs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      const text = newText.value;
+      const media = newMedia.value;
+
+      createPost({
+          title: title,
+          body: text,
+          media: media,
+        })
+      form.reset();
+      setTimeout(() => {
```
Comment (Fermain) [inline js/handlers/createPost.mjs]:
```suggestion

```

File: js/handlers/createPost.mjs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        location.reload(); 
+       }, 300)
```
Comment (Fermain) [inline js/handlers/createPost.mjs]:
```suggestion
```

File: js/handlers/getPosts.mjs [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import { renderPosts } from "../templates/posts.mjs";
```
Comment (Fermain) [inline js/handlers/getPosts.mjs]: Refactor these imports. They all come from the same file so they can all use one declaration.

File: js/handlers/login.mjs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  form.addEventListener("submit", event => {
+    event.preventDefault();
+  
+    const form = event.target;
+    const formData = new FormData(form);
+    const profile = Object.fromEntries(formData.entries());
+     
+    login(profile)
+      
+})
```
Comment (Fermain) [inline js/handlers/login.mjs]: Please format this file

File: js/handlers/updatePost.mjs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      event.preventDefault();
+
+      const form = event.target;
+      const formData = new FormData(form);
+      const post = Object.fromEntries(formData.entries())
+      post.id = id;
+
+      updatePost(post);
+      form.reset();
+      setTimeout(() => {
```
Comment (Fermain) [inline js/handlers/updatePost.mjs]: Same issue as commented on before, use `await` by turning this into an `async` function. `setTimeout` is a signal that something has gone wrong with async code.

File: js/templates/posts.mjs [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+  
+  if (avatar) {
+    const user = document.createElement("div");
+    user.classList.add("d-flex", "mt-3");
+
+    const postAuthor = document.createElement("p");
+    postAuthor.classList.add("ms-3", "mb-4");
+    postAuthor.innerHTML = name;
+
+    const UserAvatar = document.createElement("img");
```
Comment (Fermain) [inline js/templates/posts.mjs]:
```suggestion
    const userAvatar = document.createElement("img");
```

Capital letters are reserved for JS Classes

File: js/templates/posts.mjs [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+    UserAvatar.src = "/img/avatar-1606939.png";
+    UserAvatar.classList.add("ms-3", "d-flex");
+    UserAvatar.alt = "Avatar";
+    UserAvatar.height = "32";
+    user.append(UserAvatar, postAuthor)
+    postContent.append(user)
+    }
+
+
+  const date = updated.split("T");
```
Comment (Fermain) [inline js/templates/posts.mjs]:
```suggestion
  const date = new Date(updated).toLocaleDateString()
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

File: js/templates/posts.mjs [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+
+    filteredDates.forEach(i => {
+      if (i) {
+        filterText.innerHTML = "Last 24 hours";
+        parent.append(postTemplateFeed(i)) 
+      }
+    }) 
+  })
+
+  filterMoreThenTwentyFourHours.addEventListener("click", () => {
```
Comment (Fermain) [inline js/templates/posts.mjs:175]: This section could be refactored since there is a lot of repetition.

**Review by Fermain — APPROVED** [review]
This is a strong submission showing understanding and application of the course material. There are some issues that I would like you to resolve, and there is room for improvement on the UX/visual side of things.

In terms of the important part for this assessment, your code, I found it easy to read and understand - split into many small files with detailed JSDoc descriptions. I encountered no errors or major issues that prevented me from testing all of the required features.

**Pull request merged**